File: network/pi_prime_network.java
import java.net.ServerSocket;
import java.net.Socket;

public class PiPrimeNetwork {
    private ServerSocket serverSocket;

    public PiPrimeNetwork(int port) throws Exception {
        serverSocket = new ServerSocket(port);
    }

    public void listen() {
        while (true) {
            Socket clientSocket = serverSocket.accept();
            handleConnection(clientSocket);
        }
    }

    private void handleConnection(Socket clientSocket) throws Exception {
        byte[] inputBytes = new byte[16];
        clientSocket.getInputStream().read(inputBytes);
        long input = ByteBuffer.wrap(inputBytes).getLong();
        boolean output = isPrime(input);
        byte[] outputBytes = new byte[] { (byte) (output ? 1 : 0) };
        clientSocket.getOutputStream().write(outputBytes);
    }

    private boolean isPrime(long n) {
        if (n < 2) return false;
        for (long i = 2; i * i <= n; i++) {
            if (n % i == 0) return false;
        }
        return true;
    }

    public static void main(String[] args) throws Exception {
        PiPrimeNetwork network = new PiPrimeNetwork(8080);
        network.listen();
    }
}

File: network/pi_prime_network.kt
import java.net.ServerSocket
import java.net.Socket

class PiPrimeNetwork(private val port: Int) {
    private val serverSocket = ServerSocket(port)

    fun listen() {
        while (true) {
            val clientSocket = serverSocket.accept()
            handleConnection(clientSocket)
        }
    }

    private fun handleConnection(clientSocket: Socket) {
        val inputBytes = ByteArray(16)
        clientSocket.getInputStream().read(inputBytes)
        val input = ByteBuffer.wrap(inputBytes).long
        val output = isPrime(input)
        val outputBytes = byteArrayOf(if (output) 1 else 0)
        clientSocket.getOutputStream().write(outputBytes)
    }

    private fun isPrime(n: Long): Boolean {
        if (n < 2) return false
        for (i in 2..Math.sqrt(n.toDouble()).toLong()) {
            if (n % i == 0L) return false
        }
        return true
    }
}

fun main() {
    val network = PiPrimeNetwork(8080)
    network.listen()
}
